clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -4840,10 +4840,19 @@ SYCLIntegrationHeader::SYCLIntegrationHeader(bool _UnnamedLambdaSupport,
     : UnnamedLambdaSupport(_UnnamedLambdaSupport), S(_S) {}
 
 void SYCLIntegrationFooter::addVarDecl(const VarDecl *VD) {
+  // Variable template declaration can result in an error case of 'nullptr'
+  // here.
+  if (!VD)
+    return;
   // Skip the dependent version of these variables, we only care about them
   // after instantiation.
   if (VD->getDeclContext()->isDependentContext())
     return;
+
+  // Skip partial specializations of a variable template, treat other variable
+  // template instantiations as a VarDecl.
+  if (isa<VarTemplatePartialSpecializationDecl>(VD))
+    return;
```

Comment:
Just trying to understand this better.   The question was what happens after this is skipped.

Reply:
Thanks for the explanation.   It helped.
(Partly, it was a duh moment for me, as I failed to see the elimination of 'Partial' SemaTemplateInstantiateDecl.cpp :-) )